app/services/server.js

```diff
@@ -20,20 +21,47 @@ module.exports = function (options) {
 
   server.hapi.connection(server.options.connection);
 
-  // Bootstrap Hapi Server Plugins, passes the server object to the plugins.
-  require('./plugins')(server.hapi, function (err) {
-    if (err) throw err;
-  });
+  server.hapi.register(require('hapi-auth-jwt2'), err => {
+    if (err) callback(err);
 
-  server.start = function (cb) {
-    server.hapi.log(['info'], 'Database connected');
-    server.hapi.start(function () {
-      server.hapi.log(['info'], 'Server running at:' + server.hapi.info.uri);
-      if (cb && typeof cb === 'function') {
-        cb(null);
-      }
+    if (config.auth && config.auth.strategy === 'jwt') {
+      server.hapi.auth.strategy('jwt', 'jwt', true, {
```

Comment:
@danielfdsilva auth has to be initialized before all other plugins. Since plugin registration is asynchronous, unless we move to something like https://github.com/hapijs/glue, we have to do it in a callback.